pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -557,3 +560,14 @@ func trackSync(err error) {
 	}
 	endpointslicemetrics.EndpointSliceSyncs.WithLabelValues(metricLabel).Inc()
 }
+
+func dropEndpointSlicesPendingDeletion(endpointSlices []*discovery.EndpointSlice) []*discovery.EndpointSlice {
+	n := 0
+	for _, endpointSlice := range endpointSlices {
+		if endpointSlice.DeletionTimestamp == nil {
+			endpointSlices[n] = endpointSlice
+			n++
+		}
+	}
+	return endpointSlices[:n]
+}
```

Comment:
slices in go are tricky :) 

you are mutating the input, we should not return nothing, is confusing and gives the impression that is a new slice when is a point to the original slice, see https://go.dev/play/p/ICJTTE073G4

```suggestion
// dropEndpointSlicesPendingDeletion filter out the slices with DeletionTimestamp not nil
func dropEndpointSlicesPendingDeletion(endpointSlices []*discovery.EndpointSlice) {
	n := 0
	for _, endpointSlice := range endpointSlices {
		if endpointSlice.DeletionTimestamp == nil {
			endpointSlices[n] = endpointSlice
			n++
		}
	}
	endpointSlices[:n]
}
```